### PR TITLE
Internal code audit iteration one

### PR DESCRIPTION
**Pull request by hellwolf**
Walking through the code of Superfluid.sol.

Additional outputs:

- https://github.com/superfluid-finance/protocol-monorepo/wiki/About-Placeholder-Ctx

### CODE REVIEW

File: packages/ethereum-contracts/contracts/superfluid/Superfluid.sol [hunk 1]
```diff
@@ -986,12 +996,20 @@ contract Superfluid is
         internal pure
         returns (bytes memory dataWithCtx)
     {
-        // 1.a ctx needs to be padded to align with 32 bytes bundary
+        // 1.a ctx needs to be padded to align with 32 bytes boundary
         uint256 dataLen = data.length;
 
-        // double check if the ctx is a placeholder ctx
+        assert(dataLen % 32 == 0);
```
Comment (hellwolf) [inline packages/ethereum-contracts/contracts/superfluid/Superfluid.sol]: this one actually fails, will revist

Comment (hellwolf) [reply packages/ethereum-contracts/contracts/superfluid/Superfluid.sol]: Remove invalid assert
Binary data is padded, but the length field needs not

File: packages/ethereum-contracts/contracts/agreements/AgreementBase.sol [hunk 2]
```diff
@@ -5,6 +5,11 @@ import { UUPSProxiable } from "../upgradability/UUPSProxiable.sol";
 import { ISuperAgreement } from "../interfaces/superfluid/ISuperAgreement.sol";
 
 
+/**
+ * @dev The Superfluid agreement base boiler plate contract
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/agreements/AgreementBase.sol]: small fix: boilerplate is one word

File: packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol [hunk 3]
```diff
@@ -22,6 +22,15 @@ import { AgreementLibrary } from "./AgreementLibrary.sol";
 import { SlotsBitmapLibrary } from "./SlotsBitmapLibrary.sol";
 
 
+/**
+ * @dev The IConstantFlowAgreementV1 implementation
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol]: should be IInstantDistributionAgreementV1 in this file

File: packages/ethereum-contracts/contracts/agreements/AgreementBase.sol (hunk 2 above)
Comment (hellwolf) [reply packages/ethereum-contracts/contracts/agreements/AgreementBase.sol]: done

File: packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol (hunk 3 above)
Comment (hellwolf) [reply packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol]: done

File: packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol [hunk 4]
```diff
@@ -22,6 +22,15 @@ import { AgreementLibrary } from "./AgreementLibrary.sol";
 import { SlotsBitmapLibrary } from "./SlotsBitmapLibrary.sol";
 
 
+/**
+ * @dev The IInstantDistributionAgreementV1 implementation
+ *
+ * NOTE:
+ * - Please read IConstantFlowAgreementV1 for implementation notes.
```
Comment (0xdavinchee) [inline packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol]: missed this one, but everything else looks good 👍🏼

Comment (hellwolf) [reply packages/ethereum-contracts/contracts/agreements/InstantDistributionAgreementV1.sol]: 🆗

**Review by 0xdavinchee — APPROVED** [review]
looks good - took some extra time to refresh my memory on bitwise operations.

**Pull request merged**